index.js

```diff
@@ -0,0 +1,47 @@
+const mapResult = (numbers, callback) => {
+    let numbersCopy = [...numbers];
+    for (let i = 0; i < numbers.length; i++) {
+        numbersCopy[i] = callback(numbersCopy[i]);
+      }
+      return numbersCopy;
+};
+
+// const ageFinder = (array) => array <= 21;
+const filter = (array, callback) => {
+    let arrayCopy = [];
+    for (let ar of array) {
+        if (callback(ar)) arrayCopy = [...arrayCopy, ar];
+      } 
+    return arrayCopy;
+};
+// const filter = (array, callback) => {
+//     let arrayCopy = [...array];
+//     for (let i = 0; i < array.length; i++) {
+//         arrayCopy[i] = callback(arrayCopy[i]);
+//       }
+//     return arrayCopy;
+// };
+
+const findIndex = (array, callback) => {
+    for (let i = 0; i < array.length; i++) {
+        if (callback(array[i])) return i;
+    }
+
+};
+
+const reduce = (array, callback, initialValue) => {
```

Comment:
this is fine but you'll want to conditionally initialize 'i' as either 1 or zero based on the existence of the optional initialValue arg! so like: 

const reduceFunc = (array, callback, initialVal) => {   
    let accumulator;
    let i;
    if(initialVal) {
        accumulator = initialVal;
        i = 0;
    } else {
        accumulator = array[0];
        i = 1;
    }
    for(i ; i < array.length; i++) {
        accumulator = callback(accumulator, array[i])
    }
    return accumulator;
}